packages/editor-ui/src/components/NodeCreator/MainPanel.vue

```diff
@@ -0,0 +1,291 @@
+				</el-tabs>
+			</div>
+			<div v-if="searchFilter.length === 0" class="scrollable">
+				<ItemIterator
+					:elements="categorized"
+					:disabled="!!activeSubcategory"
+					:activeIndex="activeIndex"
+					:transitionsEnabled="true"
+					@selected="selected"
+				/>
+			</div>
+			<div
+				class="scrollable"
+				v-else-if="filteredNodeTypes.length > 0"
+			>
+				<ItemIterator
+					:elements="filteredNodeTypes"
+					:activeIndex="activeIndex"
+					@selected="selected"
+				/>
+			</div>
+			<NoResults v-else @nodeTypeSelected="nodeTypeSelected" />
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+
+import Vue from 'vue';
+
+import { externalHooks } from '@/components/mixins/externalHooks';
+
+import mixins from 'vue-typed-mixins';
+import ItemIterator from './ItemIterator.vue';
+import NoResults from './NoResults.vue';
+import SearchBar from './SearchBar.vue';
+import SubcategoryPanel from './SubcategoryPanel.vue';
+import { INodeCreateElement, INodeItemProps, ISubcategoryItemProps } from '@/Interface';
+import { CORE_NODES_CATEGORY  } from '@/constants';
+import SlideTransition from '../transitions/SlideTransition.vue';
+import { matchesNodeType, matchesSelectType } from './helpers';
+
+
+export default mixins(externalHooks).extend({
+	name: 'NodeCreateList',
+	components: {
+		ItemIterator,
+		NoResults,
+		SubcategoryPanel,
+		SlideTransition,
+		SearchBar,
+	},
+	props: ['categorizedItems', 'categoriesWithNodes', 'searchItems'],
+	data() {
+		return {
+			activeCategory: [] as string[],
+			activeSubcategory: null as INodeCreateElement | null,
+			activeIndex: 1,
+			nodeFilter: '',
+			selectedType: 'All',
+			searchEventBus: new Vue(),
+		};
+	},
+	computed: {
+		searchFilter(): string {
+			return this.nodeFilter.toLowerCase().trim();
+		},
+		filteredNodeTypes(): INodeCreateElement[] {
+			const nodeTypes: INodeCreateElement[] = this.searchItems;
+			const filter = this.searchFilter;
+
+			const returnData = nodeTypes.filter((el: INodeCreateElement) => {
+				const nodeType = (el.properties as INodeItemProps).nodeType;
+				return filter && matchesSelectType(el, this.selectedType) && matchesNodeType(el, filter);
+			});
+
+			setTimeout(() => {
+				this.$externalHooks().run('nodeCreateList.filteredNodeTypesComputed', {
+					nodeFilter: this.nodeFilter,
+					result: returnData,
+					selectedType: this.selectedType,
+				});
+			}, 0);
+
+			return returnData;
+		},
+
+		categorized() {
+			return this.categorizedItems && this.categorizedItems
+				.reduce((accu: INodeCreateElement[], el: INodeCreateElement) => {
+					if (
+						el.type !== 'category' &&
+						!this.activeCategory.includes(el.category)
+					) {
+						return accu;
+					}
+
+					if (!matchesSelectType(el, this.selectedType)) {
+						return accu;
+					}
+
+					if (el.type === 'category') {
+						return [...accu, {
+							...el,
+							expanded: this.activeCategory.includes(el.category),
+						}];
+					}
+
+					return [...accu, el];
```

Comment:
I like spread operator too much
I will change to concat.. seems to perform best for arrays because of some under the hood optimizations and I like to keep things immutable.